frontend/src/components/ui/chat/ChatBlock.js

```diff
@@ -1,41 +1,92 @@
-import { useState } from "react";
-
-import TextField from "@mui/material/TextField";
-import Divider from "@mui/material/Divider";
-
-import ChatList from "./ChatList";
-import Title from "./Title";
-import styles from "./ChatBlock.module.css";
-
-const DUMMY_DATA = [
-  {
-    senderId: "perborgen",
-    text: "who'll win!who'll win!who'll win!who'll win!who'll win!who'll win!who'll win!",
-  },
-  {
-    senderId: "janedoe",
-    text: "who'll win!",
-  },
-  {
-    senderId: "jimyy",
-    text: "who'll win!",
-  },
-];
-
-function ChatBlock() {
-  const[state] = useState({ chats: DUMMY_DATA });
+import { useState, useEffect, useContext } from 'react';
+
+import OutlinedInput from '@mui/material/OutlinedInput';
+import Divider from '@mui/material/Divider';
+import IconButton from '@mui/material/IconButton';
+import InputAdornment from '@mui/material/InputAdornment';
+import SendIcon from '@mui/icons-material/Send';
+
+import ChatList from './ChatList';
+import Title from './Title';
+import styles from './ChatBlock.module.css';
+
+import useInput from '../../../hooks/use-input';
+
+import UserContext from '../../../context/user-context';
+
+import isValidChat from '../../../validators/chat-validator';
+
+const ENTER_KEY_CODE = 13;
+
+function ChatBlock({ socket, roomId }) {
+  const userCtx = useContext(UserContext);
+  const [chats, setChats] = useState([]);
+
+  const {
+    value: chatMessage,
+    isValid: chatIsValid,
+    valueChangeHandler: chatChangeHandler,
+    reset: resetChat,
+  } = useInput(isValidChat);
+
+  useEffect(() => {
+    // Using useeffect hook to avoid rendering duplicate copies of chat received
+    // (Only render chat block component once after receive message)
+    socket.on('receive-chat', (senderUsername, chatMessage) => {
+      setChats((chats) => [...chats, { senderUsername, text: chatMessage }]);
+    });
+
+    // The listeners must be removed in the cleanup step,
+    // in order to prevent multiple event registrations.
+    return () => socket.off('receive-chat');
+  }, [socket]);
+
+  const sendChatHandler = (event, isMouseClick = false) => {
```

Comment:
May I suggest a refactoring: define `sendChatHandler()` to only handle the validation and emit the `send-chat` event and another method, e.g., `onKeyDownHandler()`, to check if the key pressed is the `ENTER_KEY_CODE` and if so, call the `sendChatHandler()`. This eliminates the need for the `isMouseClick` argument, which does not make much sense in the context of this method.